### PR TITLE
Check translation status of form elements

### PR DESCRIPTION
**Pull request by dsamojlenko**
# Summary | Résumé

The checkTranslated function was basically just a placeholder, only checking the translated status of a single field. This fleshes out the checkTranslated function to check the entire form.

# Test instructions | Instructions pour tester la modification

- Build a form without adding translations
- Check the Publish translate status
- Translate the form
- Check the Publish translate status again

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [ ] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [ ] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: components/form-builder/hooks/useAllowPublish.tsx [hunk 1]
```diff
@@ -1,32 +1,52 @@
...
+
+const isDescriptionTranslated = (element: Description) => {
+  if (!element.descriptionEn || !element.descriptionFr) {
+    throw MissingTranslation;
+  }
+};
+
+const isElementTranslated = (element: ElementType) => {
+  if (element.type === "richText") {
+    isDescriptionTranslated(element.properties);
```
Comment (timarney) [inline components/form-builder/hooks/useAllowPublish.tsx:21]: Can we do an early return here to get rid of the else?

File: components/form-builder/hooks/useAllowPublish.tsx [hunk 2]
```diff
@@ -1,32 +1,52 @@
...
 };
 
 const checkTranslated = (form: FormSchema) => {
   try {
-    isTranslated(form.titleEn, form.titleFr);
-    isTranslated(form.introduction.descriptionEn, form.introduction.descriptionFr);
-    isTranslated(form.privacyPolicy.descriptionEn, form.privacyPolicy.descriptionFr);
-    isTranslated(form.endPage.descriptionEn, form.endPage.descriptionFr);
+    isTitleTranslated(form);
+    isDescriptionTranslated(form.introduction);
```
Comment (timarney) [inline components/form-builder/hooks/useAllowPublish.tsx:44]: Might work not 100%
```
[form.introduction,form.privacyPolicy, form.endPage ].every(isDescriptionTranslated)
```

Comment (dsamojlenko) [reply components/form-builder/hooks/useAllowPublish.tsx:51]: that's cool

Comment (dsamojlenko) [reply components/form-builder/hooks/useAllowPublish.tsx:51]: actually, that didn't work

File: components/form-builder/hooks/useAllowPublish.tsx (hunk 1 above)
Comment (dsamojlenko) [reply components/form-builder/hooks/useAllowPublish.tsx]: refactored a bit, think it's better

**Review by timarney — APPROVED** [review]

**Pull request merged**